### ISSUE TITLE
Attempting to install azure-storage-blob results in segmentation fault

### ISSUE THREAD

**Issue by giant090860** [Azure/azure-sdk-for-python#7639]
I tried installing `azure-storage-blob` using the command provided in the [readme](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/README.md#install-the-package) however it is resulting in a segfault:
```
$ pip install azure-storage-blob --pre
Collecting azure-storage-blob
  Using cached https://files.pythonhosted.org/packages/9e/ff/6b4765c781b81f72995442b0c431038ff278aee3e186cbc4497a8f6a639b/azure_storage_blob-12.0.0b3-py2.py3-none-any.whl
Collecting cryptography>=2.1.4 (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/e6/68/50698ce24c61db7d44d93a5043c621a0ca7839d4ef9dff913e6ab465fc92/cryptography-2.7-cp27-cp27mu-manylinux1_x86_64.whl
Collecting azure-core<2.0.0,>=1.0.0b3 (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/66/e5/bae9c44414ec8c3d596bf722e8577447858cc9df0f259beca98688246f07/azure_core-1.0.0b3-py2.py3-none-any.whl
Collecting enum34>=1.0.4; python_version < "3.4" (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
Collecting msrest>=0.5.0 (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/27/b0/c34b3ea9b2ed74b800520fbefb312cdb7f05c20b8bd42e5e7662a5614f98/msrest-0.6.10-py2.py3-none-any.whl
Collecting azure-storage-nspkg<4.0.0,>=3.0.0; python_version < "3.0" (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/ba/f6/054ace7b01c6c21b3b95a83c3997f7d6539d939a2c08c4f27f779128a030/azure_storage_nspkg-3.1.0-py2.py3-none-any.whl
Collecting futures; python_version < "3.0" (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/d8/a6/f46ae3f1da0cd4361c344888f59ec2f5785e69c872e175a748ef6071cdb5/futures-3.3.0-py2-none-any.whl
Collecting typing; python_version < "3.5" (from azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/22/30/64ca29543375759dc589ade14a6cd36382abf2bec17d67de8481bc9814d7/typing-3.7.4.1-py2-none-any.whl
Collecting asn1crypto>=0.21.0 (from cryptography>=2.1.4->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/8d/c3/be192e56ab85a7b2439caf83049b9936292aaf0398038a2fe28716f18951/asn1crypto-1.0.1-py2.py3-none-any.whl
Collecting ipaddress; python_version < "3" (from cryptography>=2.1.4->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/fc/d0/7fc3a811e011d4b388be48a0e381db8d990042df54aa4ef4599a31d39853/ipaddress-1.0.22-py2.py3-none-any.whl
Collecting cffi!=1.11.3,>=1.8 (from cryptography>=2.1.4->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/8d/e9/0c8afd1579e5cf7bc0f06fbcd7cdb954cbc0baadd505973949a99337da1c/cffi-1.12.3-cp27-cp27mu-manylinux1_x86_64.whl
Collecting six>=1.4.1 (from cryptography>=2.1.4->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/73/fb/00a976f728d0d1fecfe898238ce23f502a721c0ac0ecfedb80e0d88c64e9/six-1.12.0-py2.py3-none-any.whl
Collecting azure-nspkg; python_version < "3.0" (from azure-core<2.0.0,>=1.0.0b3->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/c2/95/af354f2f415d250dafe26a5d94230558aa8cf733a9dcbf0d26cd61f5a9b8/azure_nspkg-3.0.2-py2-none-any.whl
Collecting requests>=2.18.4 (from azure-core<2.0.0,>=1.0.0b3->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/51/bd/23c926cd341ea6b7dd0b2a00aba99ae0f828be89d72b2190f27c11d4b7fb/requests-2.22.0-py2.py3-none-any.whl
Collecting certifi>=2017.4.17 (from msrest>=0.5.0->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/18/b0/8146a4f8dd402f60744fa380bc73ca47303cccf8b9190fd16a827281eac2/certifi-2019.9.11-py2.py3-none-any.whl
Collecting isodate>=0.6.0 (from msrest>=0.5.0->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/9b/9f/b36f7774ff5ea8e428fdcfc4bb332c39ee5b9362ddd3d40d9516a55221b2/isodate-0.6.0-py2.py3-none-any.whl
Collecting requests-oauthlib>=0.5.0 (from msrest>=0.5.0->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/c2/e2/9fd03d55ffb70fe51f587f20bcf407a6927eb121de86928b34d162f0b1ac/requests_oauthlib-1.2.0-py2.py3-none-any.whl
Collecting pycparser (from cffi!=1.11.3,>=1.8->cryptography>=2.1.4->azure-storage-blob)
Collecting urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (from requests>=2.18.4->azure-core<2.0.0,>=1.0.0b3->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/e0/da/55f51ea951e1b7c63a579c09dd7db825bb730ec1fe9c0180fc77bfb31448/urllib3-1.25.6-py2.py3-none-any.whl
Collecting chardet<3.1.0,>=3.0.2 (from requests>=2.18.4->azure-core<2.0.0,>=1.0.0b3->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl
Collecting idna<2.9,>=2.5 (from requests>=2.18.4->azure-core<2.0.0,>=1.0.0b3->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/14/2c/cd551d81dbe15200be1cf41cd03869a46fe7226e7450af7a6545bfc474c9/idna-2.8-py2.py3-none-any.whl
Collecting oauthlib>=3.0.0 (from requests-oauthlib>=0.5.0->msrest>=0.5.0->azure-storage-blob)
  Using cached https://files.pythonhosted.org/packages/05/57/ce2e7a8fa7c0afb54a0581b14a65b56e62b5759dbc98e80627142b8a3704/oauthlib-3.1.0-py2.py3-none-any.whl
Installing collected packages: asn1crypto, enum34, ipaddress, pycparser, cffi, six, cryptography, azure-nspkg, urllib3, certifi, chardet, idna, requests, azure-core, typing, isodate, oauthlib, requests-oauthlib, msrest, azure-storage-nspkg, futures, azure-storage-blob
Segmentation fault (core dumped)
```
I'm running Ubuntu 18.04. I ran `sudo apt-get update` `sudo apt-get upgrade` before running the pip command.

Has anyone else had this issue, or does anyone have any suggestions? If not, let me know if you need any more info from me. Thanks!

**Comment by kaerm** [comment]
@giant090860 thanks for letting us know! Someone from our team will take a look at this. cc: @rakshith91

**Comment by johanste** [comment]
This is not something that we've seen before. Based on the set of packages/wheels installed, I assume that you are using the system installed Python, correct (i.e. Python 2.7)? 

I'll try to set up a machine and see if I can reproduce the problem.

**Comment by giant090860** [comment]
@johanste 
```
$ python --version
Python 2.7.15+
```

I just noticed that interestingly, I am able to import from this package
```
>>> from azure.storage.blob import BlobClient
>>> 
```
I even successfully connected to azure and downloaded a blob from it.

So I suppose the segmentation fault is a false flag? The package is apparently working, and according to pip freeze, its installed:
```
$ pip freeze | grep blob
azure-storage-blob==12.0.0b3
```

I'll let you know if anything doesn't work, but so far, it appears to be working even though I got a seg fault during installation.

**Comment by johanste** [comment]
Ok - thanks for the update. For what it's worth, I did *not* see the segmentation fault on a clean ubuntu 18.04 box using the system install python. In addition to the steps you listed above, I also installed `sudo apt-get install python-pip` on the box after doing the update/upgrade steps. 

A quick search led me to a couple of [similar issues](https://bugs.launchpad.net/ubuntu/+source/python-pip/+bug/1821483) that may be of interest.
